specs/has_groups.md

```diff
@@ -12,7 +12,7 @@ also provide a way to actually access the contained objects.  This is done via `
 distinguished from a node by containing other objects it also represents a vertex "containing" or "linked-to" other
 vertices.  From this we formulate the requirements for this interface
 
-1. `__getitem__()` must return a non-`None` object for each string returned by `list_nodes()` and `list_groups()`.
+1. `__getitem__()` must return any or a non-`None` object for each string returned by `list_nodes()` or `list_groups()`, repectively.
```

Comment:
> must return any or a non-`None` object

Not sure what that means.

Reply:
"any of"?

Reply:
Sorry,
'any object or a non-None type object' since getitem of a node might return `None`...

Reply:
Is it better now?